pom.xml

```diff
@@ -29,13 +29,88 @@
     </license>
   </licenses>
 
+  <modules>
+    <module>arrow-data-source</module>
+    <module>native-sql-engine/core</module>
+    <module>shims</module>
+  </modules>
+
+  <profiles>
+    <profile>
+      <id>spark</id>
+      <activation>
+          <activeByDefault>true</activeByDefault>
+      </activation>
+      <properties>
+          <spark.version>${spark311.version}</spark.version>
+      </properties>
+    </profile>
+    <profile>
+        <id>spark-3.1.1</id>
+        <properties>
+            <spark.version>${spark311.version}</spark.version>
+        </properties>
+    </profile>
+    <profile>
+      <id>hadoop-2.7.4</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <properties>
+        <hadoop.version>2.7.4</hadoop.version>
+      </properties>
+    </profile>
+    <profile>
+      <id>hadoop-3.2</id>
+      <properties>
+        <hadoop.version>3.2.0</hadoop.version>
+      </properties>
+    </profile>
+    <profile>
+      <id>incremental-scala-compiler</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <properties>
+        <scala.recompile.mode>incremental</scala.recompile.mode>
+      </properties>
+    </profile>
+    <profile>
+      <id>full-scala-compiler</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+      </activation>
+      <properties>
+        <scala.recompile.mode>all</scala.recompile.mode>
+      </properties>
+    </profile>
+    <profile>
+      <id>arrow-netty</id>
+      <activation>
+        <activeByDefault>false</activeByDefault>
+      </activation>
+      <properties>
+        <arrow-memory.artifact>arrow-memory-netty</arrow-memory.artifact>
+      </properties>
+    </profile>
+    <profile>
+      <id>arrow-unsafe</id>
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <properties>
+        <arrow-memory.artifact>arrow-memory-unsafe</arrow-memory.artifact>
+      </properties>
+    </profile>
+  </profiles>
+
   <properties>
     <scala.version>2.12.10</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
     <spark.version>3.1.1</spark.version>
     <arrow.version>4.0.0</arrow.version>
     <arrow-memory.artifact>arrow-memory-netty</arrow-memory.artifact>
-    <hadoop.version>2.7.4</hadoop.version>
+    <hadoop.version>${hadoop.version}</hadoop.version>
```

Comment:
looks a bit odd to me, this should be the same as original way?

Reply:
Not the same, the issue I found is ${hadoop.version} in arrow datasource and columnar plugin didn't catch the value from parent.
The solution can help to deliver ${hadoop.version} to both sub-projects.
My testing is to use mvn help:evaluate to confirm the difference.
The original way shows ${hadoop.version} as 2.7.4 when enabling -Phadoop-3.2.

Reply:
tried the same cmd here, my hadoop version is correct

Reply:
also add the maven version limit here?
https://github.com/apache/spark/blob/master/pom.xml#L118

Reply:
Yes, we can do this by using the same maven plugin.
Let me add it and submit a new PR.